libexec/pyenv-version-file-read

```diff
@@ -11,9 +11,15 @@ if [ -s "$VERSION_FILE" ]; then
   IFS="${IFS}"$'\r'
   sep=
   while read -n 1024 -r version _ || [[ $version ]]; do
-      [[ -z $version || $version == \#* ]] && continue
-      printf "%s%s" "$sep" "$version"
-      sep=:
+    [[ -z $version || $version == \#* ]] && continue
+    # The version string is used to construct a path. Non-zero exit when
+    # invalid characters are found. This prevents issues such as CVE-2022-35861.
+    if [[ ! "$version" =~ "^[a-z0-9][-.0-9a-z]+$" ]]; then
```

Comment:
The Linux kernel allows any characters in a directory entry except `\0` and `/` (which is a path separator). So this looks overzealous to me. The Rbenv commit does it right.